doc/apiv3-documentation.apib

```diff
@@ -3390,41 +3430,306 @@ For more details and all possible responses see the general specification of [Fo
 
     [Example Form][]
 
-## Add Watcher [/api/v3/work_packages/{work_package_id}/watchers]
++ Response 403 (application/hal+json)
 
-## Add watcher [POST]
+    Returned if the work package does not exist or the client does not have sufficient permissions to see it.
+
+    **Required permission:** edit work package
+
+    *Note that you will only receive this error, if you are at least allowed to see the corresponding work package.*
+
+    + Body
+
+            {
+                "_type": "Error",
+                "errorIdentifier": "urn:openproject-org:api:v3:errors:MissingPermission",
+                "message": "You are not allowed to edit the specified work package."
+            }
+
++ Response 404 (application/hal+json)
+
+    Returned if the work package does not exist or the client does not have sufficient permissions to see it.
+
+    **Required permission:** view work package
+
+    + Body
+
+            {
+                "_type": "Error",
+                "errorIdentifier": "urn:openproject-org:api:v3:errors:NotFound",
+                "message": "The specified work package does not exist."
+            }
+
+## Watchers [/api/v3/work_packages/{work_package_id}/watchers]
+
++ Model
+    + Body
+
+            {
+                "_links": {
+                    "self": { "href": "/api/v3/work_packages/14/watchers" }
+                },
+                "total": 2,
+                "count": 2,
+                "_type": "Collection",
+                "_embedded": {
+                  "elements": [
+                    {
+                        "_type": "User",
+                        "_links": {
+                            "self": {
+                                "href": "/api/v3/users/1",
+                                "title": "John Sheppard - j.sheppard"
+                            },
+                            "lock": {
+                                "href": "/api/v3/users/1/lock",
+                                "title": "Set lock on j.sheppard"
+                                "method": "POST"
+                            },
+                            "delete": {
```

Comment:
I know this is only a slight thing and probably just my own insecurity speaking. I also think this would take more work than might be appropriate, but: As I read this link my first thought was: "And this is probably a link to remove this user as watcher from this WP" because I am looking at a collection of watchers. Immerdiately thereafter I realized that this is wrong, e.g. because it makes no sense to "lock" a watcher. But: What do you think about postfixing the link names with "User"? (i.e. "lockUser" and "deleteUser")